File: MATLAB/P01.m
flag = 1;

while flag == 1
    
    fprintf('\n');
	fprintf('########################################\n');
	fprintf('# 1. COMPARAR 3 NÚMEROS ---------- (i) #\n');
	fprintf('# 2. SALIR ----------------------- (s) #\n');
	fprintf('########################################\n');
	fprintf('\n');

	opt = input('INGRESE LA OPCIÓN QUE DESEA EJECUTAR: ','s');
	fprintf('\n');	                
    
    switch opt
        
        
       
        case 'i'
            
            num1 = input('INGRESE EL PRIMER NÚMERO: ');
            num2 = input('INGRESE EL SEGUNDO NÚMERO: ');
            num3 = input('INGRESE EL TERCER NÚMERO: ');
            fprintf('\n');
            
            if (num1 > num2 && num1 > num3) && (num1 ~= num2 && num1 ~= num3 && num2 ~= num3)

			fprintf('LA SUMA DE LOS NÚMEROS ES: %g', (num1+num2+num3));

            elseif (num2 > num1 && num2 > num3) && (num1 ~= num2 && num1 ~= num3 && num2 ~= num3)

			fprintf('LA MULTIPLICACIÓN DE LOS NÚMEROS ES: %g',(num1*num2*num3));

            elseif (num3 > num2 && num3 > num1) && (num1 ~= num2 && num1 ~= num3 && num2 ~= num3)

			fprintf('LA CONCATENACIÓN DE LOS TRES NÚMEROS ES: %g%g%g',num1,num2,num3);

            elseif (num1 == num2 && num3 ~= num1) || (num1 == num3 && num2 ~= num1) || (num2 == num3 && num1 ~= num3)

                if num1 == num2

				fprintf('EL NÚMERO DIFERENTE ES: %g', num3);
            
                elseif num1 == num3

				fprintf('EL NÚMERO DIFERENTE ES: %g', num2);
                
                else
                    
				fprintf('EL NÚMERO DIFERENTE ES: %g', num1);
                
                end
            
            else
			
			fprintf('LOS NÚMEROS SON IGUALES %g %g %g \n: ',num1,num2,num3);

            end
        case 's'
            
            flag = 0;
        
        otherwise
            
            fprintf('######################################\n');
            fprintf('# USTED NO INGRESO UNA OPCIÓN VALIDA #\n');
            fprintf('######################################\n');
    end
    
    fprintf('\n');
    
end

fprintf('####################################\n');
fprintf('# GRACIAS POR UTILIZAR EL PROGRAMA #\n');
fprintf('####################################\n');

File: PYTHON/P01.py
flag = True


while flag == True:

	try:

		archivo = open("Salida_P01.txt","a")
		print("")
		print("######################################")
		print("# 1. INGRESAR NÚMEROS ---------- (i) #")
		print("# 2. SALIR --------------------- (s) #")
		print("######################################")
		print("")

		opt = input("¿QUE OPCIÓN DESEA EJECUTAR? ")
		print("")		

		if opt == "i":
			
			num1 = int(input("INGRESE EL PRIMER NÚMERO: "))
			num2 = int(input("INGRESE EL SEGUNDO NÚMERO: "))
			num3 = int(input("INGRESE EL TERCER NÚMERO: "))
			
			print("")

			if (num1 > num2 and num1 > num3) and (num1 != num2 and num1 != num3 and num2 != num3):

				print("LA SUMA DE LOS NÚMEROS ES: ", (num1+num2+num3))
				archivo.write('LA SUMA DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' ES IGUAL A: %s'%(num1+num2+num3)+'\n')
				archivo.close()

			elif (num2 > num1 and num2 > num3) and (num1 != num2 and num1 != num3 and num2 != num3):

				print("LA MULTIPLICACIÓN DE LOS NÚMEROS ES: ",(num1*num2*num3))
				archivo.write('LA MULTIPLICACIÓN DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' ES IGUAL A: %s'%(num1*num2*num3)+'\n')
				archivo.close()

			elif (num3 > num2 and num3 > num1) and (num1 != num2 and num1 != num3 and num2 != num3):

				print("LA CONCATENACIÓN DE LOS TRES NÚMEROS ES:",(str(num1)+str(num2)+str(num3)))
				archivo.write('LA CONCATENACIÓN DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' ES IGUAL A: %s'%(str(num1)+str(num2)+str(num3))+'\n')
				archivo.close()

			elif (num1 == num2 and num3 != num1) or (num1 == num3 and num2 != num1) or (num2 == num3 and num1 != num3):

				if num1 == num2:

					print("EL NÚMERO DIFERENTE ES: ", num3);
					archivo.write('DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' EL NÚMERO DIFERENTE ES: %s'%num3+'\n')
					archivo.close()

				elif num1 == num3:

					print("EL NÚMERO DIFERENTE ES: ", num2);
					archivo.write('DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' EL NÚMERO DIFERENTE ES: %s'%num2+'\n')
					archivo.close()

				else:

					print("EL NÚMERO DIFERENTE ES: ", num1);
					archivo.write('DE LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' EL NÚMERO DIFERENTE ES: %s'%num1+'\n')
					archivo.close()

			else:
				
				print("LOS NÚMEROS SON IGUALES: ",num1,num2,num3)
				archivo.write('LOS NUMEROS %s, %s Y %s'%(num1,num2,num3)+' SON IGUALES'+'\n')
				archivo.close()


		elif opt == "s":

			flag = False

		else:

			print("######################################")
			print("# USTED NO INGRESO UNA OPCIÓN VALIDA #")
			print("######################################")

	except:

		print("USTED NO INGRESO UN NÚMERO ENTERO POSITIVO.")

print("")
print("####################################")
print("# GRACIAS POR UTILIZAR EL PROGRAMA #")
print("####################################") 